File: 0232-implement-queue-using-stacks/0232-implement-queue-using-stacks.cpp
class MyQueue {
private:
    stack<int> stack1;
    stack<int> stack2;
public:
    MyQueue() {
        
    }
    
    void push(int x) {
        if(stack1.empty()){
            stack1.push(x);
        }
        else{
            while(!stack1.empty()){
                stack2.push(stack1.top());
                stack1.pop();
            }
            
            stack1.push(x);
            
            while(!stack2.empty()){
                stack1.push(stack2.top());
                stack2.pop();
            }
        }
    }
    
    int pop() {
        int element = stack1.top();
        stack1.pop();
        return element;
    }
    
    int peek() {
        return stack1.top();
    }
    
    bool empty() {
        return stack1.empty() && stack2.empty();
    }
};

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue* obj = new MyQueue();
 * obj->push(x);
 * int param_2 = obj->pop();
 * int param_3 = obj->peek();
 * bool param_4 = obj->empty();
 */

File: 0232-implement-queue-using-stacks/0232-implement-queue-using-stacks.java
class MyQueue {
    private Stack<Integer> stack1;
    private Stack<Integer> stack2;
    
    public MyQueue() {
        stack1 = new Stack<>();
        stack2 = new Stack<>();
    }
    
    public void push(int x) {
        if(stack1.isEmpty()){
            stack1.add(x);
        }
        else{
            while(!stack1.isEmpty()){
                stack2.add(stack1.peek());
                stack1.pop();
            }
            
            stack1.add(x);
            
            while(!stack2.isEmpty()){
                stack1.add(stack2.peek());
                stack2.pop();
            }
        }
    }
    
    public int pop() {
        int element = stack1.peek();
        stack1.pop();
        return element;
    }
    
    public int peek() {
        return stack1.peek();
    }
    
    public boolean empty() {
        return stack1.isEmpty() && stack2.isEmpty();
    }
}

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue obj = new MyQueue();
 * obj.push(x);
 * int param_2 = obj.pop();
 * int param_3 = obj.peek();
 * boolean param_4 = obj.empty();
 */